engine/storage/snapshot/src/org/apache/cloudstack/storage/snapshot/XenserverSnapshotStrategy.java

```diff
@@ -374,10 +374,14 @@ public SnapshotInfo takeSnapshot(SnapshotInfo snapshot) {
                     throw new CloudRuntimeException(result.getResult());
                 }
             } finally {
-                if (result != null && result.isSuccess()) {
-                    volumeInfo.stateTransit(Volume.Event.OperationSucceeded);
-                } else {
-                    volumeInfo.stateTransit(Volume.Event.OperationFailed);
+                List<SnapshotVO> activeVolumeSnapshots = snapshotDao.listByStatus(snapshot.getVolumeId(), Snapshot.State.Creating);
+                if (activeVolumeSnapshots.isEmpty()) {
```

Comment:
@kishankavala The active snapshots jobs for that volume will be checked here from the snapshots table. Lock is acquired on snapshots table when this is performed.